firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/I2C/HoneywellABP/Device.cpp

```diff
@@ -10,39 +10,35 @@
  *    Arduino library by ij96 (https://github.com/ij96)
  */
 
-#include "Pufferfish/Driver/I2C/HoneywellABP.h"
+#include "Pufferfish/Driver/I2C/HoneywellABP/Device.h"
 
 #include <array>
 #include <climits>
 
-namespace Pufferfish::Driver::I2C {
+#include "Pufferfish/Util/Endian.h"
+#include "Pufferfish/Util/Ranges.h"
 
-I2CDeviceStatus HoneywellABP::read_sample(ABPSample &sample) {
+namespace Pufferfish::Driver::I2C::HoneywellABP {
+
+I2CDeviceStatus Device::read_sample(ABPSample &sample) {
   std::array<uint8_t, 2> data{{0, 0}};
   I2CDeviceStatus ret = dev_.read(data.data(), data.size());
   if (ret != I2CDeviceStatus::ok) {
     return ret;
   }
 
-  static const uint8_t status_shift = 6;
-  static const size_t bridge_high = 0;
-  static const size_t bridge_low = 0;
   static const uint16_t bridge_mask = 0x3FFF;
   sample.status = ABPStatus(data[0] >> status_shift);
-  sample.bridge_data = (data[bridge_high] << static_cast<uint8_t>(CHAR_BIT)) + data[bridge_low];
+  Util::read_bigend(data.data(), sample.bridge_data);
   sample.bridge_data &= bridge_mask;
   sample.pressure = raw_to_pressure(sample.bridge_data);
   sample.unit = unit;
 
   return I2CDeviceStatus::ok;
 }
 
-float HoneywellABP::raw_to_pressure(uint16_t output) const {
-  if (output < output_min) {
-    output = output_min;
-  } else if (output > output_max) {
-    output = output_max;
-  }
+float Device::raw_to_pressure(uint16_t output) const {
+  Util::clamp<uint16_t>(output, output_min, output_max);
```

Comment:
Does `Util::clamp` produce the clamped result as its returned value, or does it modify the first parameter as an output parameter?

Reply:
Oops my bad! it returns the clamed result.